auth-table-based-service/src/test/java/io/stargate/auth/SourceAPITest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.auth;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class SourceAPITest {
+
+    @Nested
+    class ToCustomPayload {
+
+
+        @Test
+        public void happyPath() {
+            Map<String, ByteBuffer> map = new HashMap<>();
+
+            SourceAPI.GRAPHQL.toCustomPayload(map);
+
+            assertThat(map).containsKey("stargate.sourceAPI");
+        }
+
+        @Test
+        public void nullProtected() {
+            SourceAPI.GRAPHQL.toCustomPayload(null);
+        }
+
+    }
+
+    @Nested
+    class FromCustomPayload {
+
+
+        @Test
+        public void happyPath() {
+            Map<String, ByteBuffer> map = new HashMap<>();
+            SourceAPI.GRAPHQL.toCustomPayload(map);
+
+            SourceAPI result = SourceAPI.fromCustomPayload(map, SourceAPI.CQL);
+
+            assertThat(result).isEqualTo(SourceAPI.GRAPHQL);
+        }
+
+        @Test
+        public void emptyMap() {
```

Comment:
I know it's implicit but maybe add a comment stating this verifies that our default SourceAPI is CQL?